tests/Integration/Behaviour/Features/Context/Domain/ApiAccessManagementFeatureContext.php

```diff
@@ -0,0 +1,172 @@
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\Query\GetApiAccessForEditing;
+use PrestaShop\PrestaShop\Core\Domain\ApiAccess\ValueObject\ApiAccessId;
+use Tests\Integration\Behaviour\Features\Context\CommonFeatureContext;
+use Tests\Integration\Behaviour\Features\Context\Util\DataComparator;
+use Tests\Integration\Behaviour\Features\Context\Util\DataTransfer;
+use Tests\Resources\Resetter\ApiAccessResetter;
+
+class ApiAccessManagementFeatureContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When /^I create an api access "(.+)" with following properties:$/
+     */
+    public function createApiAccessUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $data = $this->fixDataType($table->getRowsHash());
+        $commandBus = $this->getCommandBus();
+
+        $command = new AddApiAccessCommand(
+          $data['clientName'],
+          $data['apiClientId'],
+          $data['enabled'],
+          $data['description']
+        );
+
+        try {
+            /** @var ApiAccessId $id */
+            $id = $commandBus->handle($command);
+
+            $this->getSharedStorage()->set($apiAccessReference, $id->getValue());
+        } catch (ApiAccessConstraintException $e) {
+            $this->setLastException($e);
+        }
+    }
+
+    /**
+     * @Then /^api access "(.+)" should have the following properties:$/
+     */
+    public function assertQueryApiAccessProperties($apiAccessReference, TableNode $table)
+    {
+        $expectedData = $table->getRowsHash();
+        $this->getSharedStorage()->exists($apiAccessReference);
+
+        $queryBus = $this->getQueryBus();
+        $result = $queryBus->handle(new GetApiAccessForEditing($this->getSharedStorage()->get($apiAccessReference)));
+
+        $serializer = CommonFeatureContext::getContainer()->get('serializer');
+        $realData = $serializer->normalize($result);
+
+        DataComparator::assertDataSetsAreIdentical($expectedData, $realData);
+    }
+
+    /**
+     * @When /^I edit api access "(.+)" with the following values:$/
+     */
+    public function editCustomerUsingCommand($apiAccessReference, TableNode $table)
+    {
+        $this->getSharedStorage()->exists($apiAccessReference);
+        $data = $table->getRowsHash();
+
+        $commandBus = $this->getCommandBus();
+
+        $command = new EditApiAccessCommand($this->getSharedStorage()->get($apiAccessReference));
+        $data = $this->fixDataType($data);
+
+        DataTransfer::transferAttributesFromArrayToObject($data, $command);
+        $commandBus->handle($command);
+    }
+
+    /**
+     * @Then I should get an error that :fieldName is not unique
+     */
+    public function iShouldGetAnErrorThatFieldIsNotUnique(string $fieldName): void
+    {
+        $this->assertLastErrorIs(
+            ApiAccessConstraintException::class,
+            $this->getUnicityConstraintErrorCode($fieldName)
+        );
+    }
+
+    /**
+     * @Then I should get an error that :fieldName is blank
```

Comment:
```suggestion
     * @Then I should get an error that :fieldName is invalid
```

Not blank is not the only constraint that is checked and can make a field invalid, there's also the max length so the naming should be more generic here